### ISSUE TITLE
Increase momentum on interventions in general

### ISSUE THREAD

**Issue by wisniewskit** [mozilla/webcompat-team-okrs#235]
- post-refactor (#231), we should have an easier time adding and maintaining site-patches
- we should seek other avenues for reducing friction on site-patches

**Comment by karlcow** [comment]
@wisniewskit How long do you think this should take.

**Comment by karlcow** [comment]
Or maybe the terms are currently too vague to define a plan for work,

**Comment by wisniewskit** [comment]
Yes, this is too vague as written to give a proper estimate. We can discuss it together with @denschub and @ksy36 to get more thoughts on what we'd like to do here.

**Comment by denschub** [comment]
So, this is my very personal opinion.

> post-refactor, we should have an easier time adding and maintaining site-patches

I'm not sure if that's actually true. While I think that the suggested refactoring is not a bad idea, I don't see any significant friction caused by the current state in my work with the addon. For me, while building interventions, the things that take time are

- figuring out which issues are worth an intervention
- doing full diagnosis on an issue - because we need to understand it first
- finding a way how to fix it. i.e., finding out which CSS rules to write, or which JS objects to shim
- testing if the intervention actually works.

After that, adding the intervention itself is pretty much just copy-pasting a block in a JS file, changing a few values, creating a new JS/CSS file, and hitting save. Even testing the intervention is relatively simple, as there is automation for that. For CSS interventions, I usually use the devtools style tab and write a bit of CSS until the issue is fixed. When I figured it out, the process from there takes no more than 5 minutes at most, for me.

So the main source of friction here is the requirement to a) spend more time on diagnosis, b) spend the time to figure out how to intervene. Unfortunately, this does not translate into tangible tasks to drive improvements, let alone predictable engineering tasks.

It is very possible that I'm not seeing issues, though, mainly because I built the basis and I've got very very used to it. Occasionally, if you're that deep into it, you're getting really good at not noticing issues. :) So Ksenias and Karls opinion is worth more in this case.
